conductor-client/src/main/kotlin/com/openlattice/graph/Graph.kt

```diff
@@ -977,6 +977,38 @@ class Graph(
         }.toSet()
     }
 
+    override fun getNeighborEdgeEntitySets(
+            entitySetIds: Set<UUID>,
+            entityKeyIds: Set<UUID>? ): Set<UUID> {
+
+        val srcEntityKeyIdFilter = if (entityKeyIds == null) "" else "AND ${SRC_ENTITY_KEY_ID.name} = ANY(?)"
+        val dstEntityKeyIdFilter = if (entityKeyIds == null) "" else "AND ${DST_ENTITY_KEY_ID.name} = ANY(?)"
+
+        val query = """
+            SELECT DISTINCT ${EDGE_ENTITY_SET_ID.name} FROM ${E.name}
+            WHERE ( ${DST_ENTITY_SET_ID.name} = ANY(?) $dstEntityKeyIdFilter )
+            OR ( ${SRC_ENTITY_SET_ID.name} = ANY(?) $srcEntityKeyIdFilter )
+        """.trimIndent()
+
+        val reader = dataSourceResolver.getDefaultDataSource()
+        return BasePostgresIterable(PreparedStatementHolderSupplier(reader, query) { ps ->
+            val entitySetIdsArr = PostgresArrays.createUuidArray(ps.connection, entitySetIds)
+            var index = 0
```

Comment:
No change required, just a thought on structure. I think it's more straightforward to initialize index to 1 and do `index++` (or maybe even skip it altogether).